cirq_superstaq/compiler_output.py

```
@@ -132,7 +132,7 @@ def read_json_ibmq(json_dict: Dict[str, Any], circuits_is_list: bool) -> Compile
 
 
 def read_json_aqt(  # pylint: disable=missing-param-doc
-    json_dict: Dict[str, Any], circuits_is_list: bool, num_eca_circuits: int = 0
+    json_dict: Dict[str, Any], circuits_is_list: bool, num_eca_circuits: Optional[int] = None
```

Why this change?

---

`_aqt_compile()` always passes a `num_eca_circuits` argument to `read_json_aqt()` regardless of whether it's called from `aqt_compile()` and `aqt_compile_eca()`, so it seemed a lot more clear to have it pass the default `None` for non-eca compilation instead of having it pass 0 and treating that as a kind of magic number

(your comment also made me realize i forgot to propagate this change through the rest of the function; fixed now)

---

though now i'm thinking we should also raise an error in `aqt_compile_eca` if you pass `num_equivalent_circuits=0`

---

> though now i'm thinking we should also raise an error in `aqt_compile_eca` if you pass `num_equivalent_circuits=0`

will do this in a follow-up pr

---

just filed https://github.com/SupertechLabs/general-superstaq/issues/227 (with universal tag)